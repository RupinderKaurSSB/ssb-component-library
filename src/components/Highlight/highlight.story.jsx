import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import Highlight from './index';

const negativeStyle = {
	backgroundColor: '#274247',
	padding: '100px',
};

storiesOf('Highlight', module).addDecorator(centered)
	.add('Default', () => (
		<div style={{ padding: '100px' }}>
			<p>
				«Ser du etter en hytte med slik beliggenhet, må du regne med å måtte ut med en bra slump med penger
				<Highlight>
					hvis du ikke er så heldig å arve
				</Highlight>
				– selv om også dét kan være dyrt.»
			</p>
		</div>
	))
	.add('Negative', () => (
		<div style={negativeStyle}>
			<p>
				«Ser du etter en hytte med slik beliggenhet, må du regne med å måtte ut med en bra slump med penger
				<Highlight negative>
					hvis du ikke er så heldig å arve
				</Highlight>
				– selv om også dét kan være dyrt.»
			</p>
		</div>
	));
