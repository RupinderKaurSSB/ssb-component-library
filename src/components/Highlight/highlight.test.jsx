import React from 'react';
import {shallow} from 'enzyme';
import Highlight from './index';

describe('Highlight component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<Highlight>«Highlight»</Highlight>);
		expect(wrapper).toMatchSnapshot();
	});
	test('Toggles classNames correctly', () => {
		const wrapper = shallow(<Highlight negative>«Highlight»</Highlight>);
		expect(wrapper.hasClass('negative')).toEqual(true);
	});

});
