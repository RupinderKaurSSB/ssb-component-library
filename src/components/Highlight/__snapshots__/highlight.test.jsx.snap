// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight component Matches the snapshot 1`] = `
<mark
  className="ssb-mark"
>
  «Highlight»
</mark>
`;
